view/frontend/web/js/view/payment/method-renderer/adyen-cc-method.js

```diff
@@ -335,17 +337,19 @@ define(
                     event.preventDefault();
                 }
 
-                if (this.validate() && additionalValidators.validate()) {
-                    fullScreenLoader.startLoader();
-                    self.isPlaceOrderActionAllowed(false);
+                setBillingAddressAction(globalMessageList).then(function() {
```

Comment:
the placeOrder does already setting the billingAddress does this not execute this logic twice ?
This will decrease performance as 2 api calls are needed on the place order button. Would it not be better to pass the billingAddress into the API call ? as well maybe better for supporting PWA apps ?

Reply:
Yes the original placeOrder flow will update it again but only after the adyen /payments call is done with the old billing address. Why I choose this solution is because even if we get redirect (3ds1) back from Adyen we save the billing address in the quote and if something went wrong the user will still have the new billing address already saved. 

What I can do to optimise it is to update the quote in our custom controller before making the /payments call and then we have 1 less ajax call but the same functionality. 
I will do the changes soon.